tests/plugins/easyimage/pasteintegration.js

```diff
@@ -0,0 +1,94 @@
+	};
+	// jscs:enable maximumLineLength
+
+	bender.editor = true;
+
+	function assertEasyImageUpcast( config ) {
+		var editor = config.editor,
+			pastedData = config.pastedData,
+			shouldUpcast = config.shouldUpcast,
+			// spy checks if paste listener in easyimage plugin activates an early return,
+			// by spying the method available after the early return statement.
+			spy = sinon.spy( editor.widgets.registered.easyimage, '_spawnLoader' );
```

Comment:
I had also some doubts about it. Unfortunately, I couldn't find any good way to spy on the listener itself and checking the result of the return expression. As there might be multiple paste listeners.
So I remain with an option which checks what is a result in the editor together with checking if function inside the listener is executed or not.
I can remove the `spy` if you think it might be a problem in the future.

Reply:
Ok, let's leave it as is 👍